fvm/evm/handler/handler.go

```diff
@@ -0,0 +1,337 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/fvm/evm/types"
+)
+
+// ContractHandler is responsible for triggering calls to emulator, metering,
+// event emission and updating the block
+//
+// TODO and Warning: currently database keeps a copy of roothash, and if after
+// commiting the changes by the evm we want to revert in this code we need to reset that
+// or we should always do all the checks and return before calling the emulator,
+// after that should be only event emissions and computation usage updates.
+// thats another reason we first check the computation limit before using.
+// in the future we might benefit from a view style of access to db passed as
+// a param to the emulator.
+type ContractHandler struct {
+	blockchain        types.BlockChain
+	backend           types.Backend
+	emulator          types.Emulator
+	lastExecutedBlock *types.Block
+	uuidIndex         uint64
+	totalSupply       uint64
+}
+
+var _ types.ContractHandler = &ContractHandler{}
+
+func NewContractHandler(
+	blockchain types.BlockChain,
+	backend types.Backend,
+	emulator types.Emulator,
+) *ContractHandler {
+	lastExecutedBlock, err := blockchain.LatestBlock()
+	// fatal error
+	if err != nil {
+		panic(err)
```

Comment:
Are these panics going to get intercepted, this should probably not reach the block computer.

Reply:
this is a fatal one, should stop the node while executing a transaction but not during script execution. :-?

Reply:
currently the block computer and query executor do not know how to deal with panics that well... 
But returning an error here would probably result in the expected behaviour, because the error is not a coded error and would get interpreted as a failure

Reply:
these are actually intercepted by Cadence runtime.